src/OrleansRuntime/GrainDirectory/GrainDirectoryPartition.cs

```diff
@@ -40,6 +40,8 @@ internal class GrainInfo : IGrainInfo
     {
         public Dictionary<ActivationId, IActivationInfo> Instances { get; private set; }
         public int VersionTag { get; private set; }
+
+        // TODO the following field creates memory overhead for ALL grains in directory, perhaps it could be done differently
         public bool SingleInstance { get; private set; }
```

Comment:
what does this comment mean? This property was here all along and it was not added by you